File: other/test/BSTNode.java
public class BSTNode {

    private int value_;
    private BSTNode parent_ = null;
    private BSTNode left_ = null;
    private BSTNode right_ = null;

    public BSTNode (int val){
        this.value_ = val;
    }

    //Getters
    public BSTNode getParent(){ return this.parent_;}
    public BSTNode getLeftChild(){ return this.left_;}
    public BSTNode getRightChild(){ return this.right_;}
    public int getValue(){ return this.value_;}

    //Setters
    public void setParent(BSTNode n){ this.parent_ = n;}
    public void setLeftChild(BSTNode n){ this.left_ = n;}
    public void setRightChild(BSTNode n){ this.right_ = n;}
    public void setValue(int i){ this.value_ = i;}

}

File: other/test/BSTNode.py

class BSTNode:

    value = 0
    parent = None 
    left = None 
    right = None 

    def __init__(self, val):
        self.value = val

    def getParent(self):
        return self.parent
    
    def getLeftChild(self):
        return self.left
    
    def getRightChild(self):
        return self.right

    def setParent(self, val):
        self.parent = val

    def setLeftChild(self, val):
        self.left = val;

    def setRightChild(self, val):
        self.right = val;

        
